includes/abstracts/abstract-wc-order.php

```diff
@@ -780,14 +780,36 @@ public function get_item_count( $item_type = '' ) {
 	}
 
 	/**
-	 * Get an order item object, based on it's type.
+	 * Get an order item object, based on it's ID. The item must belong to the current
+	 * order. If the item cannot be found or it doens't belong to current order
+	 * FALSE will be returned.
 	 *
 	 * @since  3.0.0
 	 * @param  int $item_id
 	 * @return WC_Order_Item
 	 */
 	public function get_item( $item_id ) {
-		return WC_Order_Factory::get_order_item( $item_id );
+		foreach ( $this->items as $items ) {
```

Comment:
@mikejolley Note that the object returned by `get_item` should be consistent with the one included in the array that is returned by `get_items`. 

e.g. Product Bundles hooks into `woocommerce_order_get_items` to modify what's exported to Shipstation, depending on whether a bundled item is "assembled" or not in its container item. In this case, totals and weights are aggregated into the container item.

To ensure consistency between `get_items` and `get_item`, at the very least, the object returned by `get_item` should be filtered. Otherwise, `get_item` should call `get_items`, and `get_items` should index the returned objects by item id (it doesn't at the moment).